pkg/platform/runtime/setup/setup.go

```diff
@@ -626,8 +626,15 @@ func (s *Setup) fetchAndInstallArtifactsFromBuildPlan(installFunc artifactInstal
 	}
 
 	if s.target.ProjectDir() != "" && s.cfg.GetBool(constants.OptinBuildscriptsConfig) {
-		if err := buildscript.Update(s.target, buildResult.AtTime, buildResult.BuildExpression, s.auth); err != nil {
-			return nil, nil, errs.Wrap(err, "Could not save build script.")
+		script, err := buildscript.NewFromCommit(buildResult.AtTime, buildResult.BuildExpression)
+		if err != nil {
+			return nil, nil, errs.Wrap(err, "Could not construct build script")
+		}
+		if err := s.store.StoreBuildScript(script); err != nil {
+			return nil, nil, errs.Wrap(err, "Could not save build script")
+		}
+		if err := buildscript.Update(s.target, script.AtTime, script.Expr, s.auth); err != nil {
+			return nil, nil, errs.Wrap(err, "Could not update build script.")
```

Comment:
This feels like it's gonna be a pain to maintain if we just have to remember to do this anytime we update a buildscript. I suggest we make this its own ticket and we leave the opt-in case unaddressed for this story.

Reply:
I filed DX-2650 for this.